controller/src/test/java/org/jboss/as/controller/persistence/ConfigurationFileUnitTestCase.java

```diff
@@ -0,0 +1,78 @@
+package org.jboss.as.controller.persistence;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+/**
+ *
+ * @author <a href="padamec@redhat.com">Petr Adamec</a>
+ */
+public class ConfigurationFileUnitTestCase {
+    private static final String NAME = "standalone.xml";
+    private static final String CONFIGURATION_DIR = "tmp_conf";
+    private static final String HISTORY_DIRECTORY = "standalone_xml_history";
+    private static final String SNAPSHOT_DIRECTORY = "snapshot";
+
+    File configurationDir;
+    File configurationFile;
+    File historyDirectory;
+    File snapshotsDirectory;
+
+    @Before
+    public void createDirectoryAndFile() throws Exception {
+        configurationDir = createDir(null,CONFIGURATION_DIR);
+        historyDirectory = createDir(configurationDir, HISTORY_DIRECTORY);
+        snapshotsDirectory = createDir(historyDirectory, SNAPSHOT_DIRECTORY);
+        configurationFile = createFile(snapshotsDirectory, NAME, "");
+    }
+
+    private File createDir(File parent, String name) {
+        File created = new File(parent, name);
+        created.mkdir();
+        if (!created.exists()) {
+            Assert.fail("Could not create " + created);
+        }
+        return created;
+    }
+
+    private File createFile(File dir, String name, String contents) throws IOException {
+        File file = new File(dir, name);
+        if (contents != null) {
+            Files.write(file.toPath(), contents.getBytes(StandardCharsets.UTF_8));
+        }
+        return file;
+    }
+
+    @Test
+    public void testConstructor(){
+        try{
```

Comment:
Just curious, does this pass checkstyle? Not sure if wf-core has a rule to enforce a space before/after {}. If there's no such rule then OK I guess.